src/batchUploader.ts

```diff
@@ -172,29 +200,52 @@ export class BatchUploader {
     private async prepareAndUpload(triggerFuture: boolean, useBeacon: boolean) {
         const currentUser = this.mpInstance.Identity.getCurrentUser();
 
-        const currentEvents = this.pendingEvents;
+        // const currentEvents = this.pendingEvents;
```

Comment:
This section would benefit most from a visual flow of how an event gets uploaded, and if it is in memory, local storage, when it's removed, etc.  some of the TODOs and in memory are difficult to grok without that flow chart